### ISSUE TITLE
Absent private master GPG key causes crash

### ISSUE THREAD

**Issue by kron4eg** [mssun/passforios#295]
In continuation of https://github.com/mssun/passforios/issues/205#issuecomment-519277716 I'll describe what happened there.

A bit fancy key (Curve25519) look like this:

```
$ gpg -K --with-keygrip
------------------------------------------
sec   ed25519/0x298B871AA1F5FF35 2019-08-08 [C]
      8B1A2AC11D7E587276404BF9298B871AA1F5FF35
      Keygrip = 9CA02D4D0F04324FA84E11DEA6B86A16B3E9B134
uid                   [ultimate] Pass For iOS Test User <passforios@example.com>
ssb   cv25519/0x161245A20E26C97A 2019-08-08 [E]
      Keygrip = 1016FD37E593F552520CAE111DC2510E8D28A77A
```

Then, we export only secrets for subkeys:
```
$ gpg --armor --export-secret-subkeys 0x298B871AA1F5FF35 > gpg_key
$ gpg --armor --export 0x298B871AA1F5FF35 > gpg_key.pub
```

This removes master private key out of the equation (master private key is not required for encryption/decryption to work, its only role is to `Certify`). In GPG it looks like this:

```
$ gpg -K
------------------------------------------
sec#  ed25519/0x298B871AA1F5FF35 2019-08-08 [C]
      8B1A2AC11D7E587276404BF9298B871AA1F5FF35
uid                   [ultimate] Pass For iOS Test User <passforios@example.com>
ssb   cv25519/0x161245A20E26C97A 2019-08-08 [E]
```

Notice the `sec#`. It means that there is no master private key.

Next, import resulted files via iTunes to Passforios.

Once opened in the app, app instantly crashes 1 time and on restart proceed to complain about "please setup your key".

---

But when I import whole key, including master's private key — everything suddenly works.

---

Summary: 
It's very recommended to keep your master's private key OFFLINE, but not possible in pass for ios currently.

**Comment by kron4eg** [comment]
seems like duplicate of https://github.com/mssun/passforios/issues/285
should I close this issue?

**Comment by mssun** [comment]
Hi @kron4eg, does the app crash in the latest TestFlight? In recent several PRs, we can support subkey.

**Comment by kron4eg** [comment]
@mssun sorry, I don't have passforios in my TestFlight, I think I need a link or direct invitation.

**Comment by kron4eg** [comment]
OK, sent you an email.

**Comment by kron4eg** [comment]
@mssun nope, seems like it got even worse. App crashes instantly after I'm trying to open Settings.

Steps to repro:
```
gpg --armor --export testuser | qrencode -o public.png
gpg --armor --export-secret-subkeys testuser | qrencode -o private.png
```

Import GPG keys (without master secret). App starts to crash. Sent TestFlight report.

**Comment by mssun** [comment]
Thanks @kron4eg, this is very helpful. Our PGP library is still not reliable, which is very disappointed.

**Comment by mikkeloscar** [comment]
I have a somewhat similar problem I expect. I have added subkeys, but when I try to open a password the app just crashes without any information.

I will ask to be invited to the testflight beta.

**Comment by Vintodrimmer** [comment]
Have exactly the same problem.
Reinstalling the app didn't work.
The database is empty now, but it keeps crashing every time I enter settings.
This includes the TestFlight version.

**Comment by mssun** [comment]
@Vintodrimmer, we have tests for RSA2048, RSA2048_SUB, RSA4096, RSA4096_SUB, ED25519, ED25519_SUB (https://github.com/mssun/passforios/blob/master/passKitTests/Crypto/CryptoFrameworkTest.swift#L38). All works now. Can you tell me your key information? How can I generate the key which can make the app crash or something.
